projects/lexical_shared/lib/lexical/document/store.ex

```diff
@@ -258,7 +261,15 @@ defmodule Lexical.Document.Store do
 
   @spec open(Lexical.uri(), String.t(), pos_integer()) :: :ok | {:error, :already_open}
   def open(uri, text, version) do
-    GenServer.call(name(), {:open, uri, text, version})
+    language_id = Document.language_id_from_uri(uri)
+
+    GenServer.call(name(), {:open, uri, text, version, language_id})
```

Comment:
I think having a default language id of nil will simplify things. Then we can have a three arity and four arity open function, and push determining language id down to the document, which would then not have to expose its `language_id_from_*` functions.

Reply:
Done as well.